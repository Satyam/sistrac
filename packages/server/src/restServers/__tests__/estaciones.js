import {
  MysqlError,
  initRoutes,
  testREST,
  GET,
  POST,
  PUT,
  DELETE,
} from '../../utils/testUtils';

import {
  OK,
  NOT_FOUND,
  NO_CONTENT,
  CONFLICT,
  CREATED,
} from '../../utils/httpStatusCodes';

describe('restServers/estaciones', () => {
  beforeAll(() => {
    initRoutes();
  });
  describe('/', () => {
    it(
      'should provide a list of estaciones',
      testREST({
        method: GET,
        path: '/estaciones/',
        queryResult: [{ id: 1, dato: 'uno' }, { id: 2, dato: 'dos' }],
        statusCode: OK,
        restResult: [{ id: 1, dato: 'uno' }, { id: 2, dato: 'dos' }],
      }),
    );
  });
  describe('get /RET', () => {
    it(
      'success',
      testREST({
        method: GET,
        path: '/estaciones/RET',
        queryResult: [{ idEstacion: 'RET', nombre: 'Retiro' }],
        statusCode: OK,
        restResult: { idEstacion: 'RET', nombre: 'Retiro' },
      }),
    );
    it(
      'fail',
      testREST({
        method: GET,
        path: '/estaciones/XXX',
        queryResult: [],
        statusCode: NOT_FOUND,
      }),
    );
  });
  describe('get /nombre/nombre', () => {
    it(
      'success',
      testREST({
        method: GET,
        path: '/estaciones/nombre/Retiro',
        queryResult: [{ idEstacion: 'RET', nombre: 'Retiro' }],
        statusCode: OK,
        restResult: { idEstacion: 'RET', nombre: 'Retiro' },
      }),
    );
    it(
      'fail',
      testREST({
        method: GET,
        path: '/estaciones/nombre/Retiro',
        queryResult: [],
        statusCode: NOT_FOUND,
      }),
    );
  });
  describe('get /idEstacion', () => {
    it(
      'success',
      testREST({
        method: GET,
        path: '/estaciones/RET',
        queryResult: [{ idEstacion: 'RET', nombre: 'Retiro' }],
        statusCode: OK,
        restResult: { idEstacion: 'RET', nombre: 'Retiro' },
      }),
    );
    it(
      'fail',
      testREST({
        method: GET,
        path: '/estaciones/XXX',
        queryResult: [],
        statusCode: NOT_FOUND,
      }),
    );
  });
  describe('get /existe/nombre', () => {
    it(
      '/Retiro success',
      testREST({
        method: GET,
        path: '/estaciones/existe/nombre/Retiro',
        queryResult: [{ hay: 1 }],
        statusCode: OK,
      }),
    );
    it(
      '/YYYY fail',
      testREST({
        method: GET,
        path: '/estaciones/existe/nombre/YYYY',
        queryResult: [{ hay: 0 }],
        statusCode: NOT_FOUND,
      }),
    );
  });
  describe('get /existe/ id', () => {
    it(
      '/RET success',
      testREST({
        method: GET,
        path: '/estaciones/existe/RET',
        queryResult: [{ hay: 1 }],
        statusCode: OK,
      }),
    );
    it(
      '/YYY fail',
      testREST({
        method: GET,
        path: '/estaciones/existe/YYY',
        queryResult: [{ hay: 0 }],
        statusCode: NOT_FOUND,
      }),
    );
  });
  describe('put', () => {
    it(
      '/RET success',
      testREST({
        method: PUT,
        path: '/estaciones/RET',
        queryResult: { affectedRows: 1 },
        statusCode: NO_CONTENT,
        body: {
          nombre: 'Retiro',
        },
      }),
    );
    it(
      '/XXX fail',
      testREST({
        method: PUT,
        path: '/estaciones/XXX',
        queryResult: { affectedRows: 0 },
        statusCode: NOT_FOUND,
        body: {
          nombre: 'Retiro',
        },
      }),
    );
  });
  describe('post', () => {
    it(
      'success',
      testREST({
        method: POST,
        path: '/estaciones/',
        queryResult: { affectedRows: 1 },
        statusCode: CREATED,
      }),
    );
    it(
      'duplicate',
      testREST({
        method: POST,
        path: '/estaciones/',
        queryResult: new MysqlError('ER_DUP_ENTRY'),
        statusCode: CONFLICT,
      }),
    );
  });
  describe('delete', () => {
    it(
      '/RET success',
      testREST({
        method: DELETE,
        path: '/estaciones/RET',
        queryResult: { affectedRows: 1 },
        statusCode: NO_CONTENT,
      }),
    );
    it(
      '/XXX fail',
      testREST({
        method: DELETE,
        path: '/estaciones/XXX',
        queryResult: { affectedRows: 0 },
        statusCode: NOT_FOUND,
      }),
    );
  });
});
