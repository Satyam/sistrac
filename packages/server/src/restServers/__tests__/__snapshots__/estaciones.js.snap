// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restServers/estaciones / should provide a list of estaciones 1`] = `Object {}`;

exports[`restServers/estaciones / should provide a list of estaciones 2`] = `"select * from Estaciones order by nombre"`;

exports[`restServers/estaciones delete /RET success 1`] = `Object {}`;

exports[`restServers/estaciones delete /RET success 2`] = `"delete from Estaciones where idEstacion = 'RET'"`;

exports[`restServers/estaciones delete /XXX fail 1`] = `Object {}`;

exports[`restServers/estaciones delete /XXX fail 2`] = `"delete from Estaciones where idEstacion = 'XXX'"`;

exports[`restServers/estaciones get /RET fail 1`] = `Object {}`;

exports[`restServers/estaciones get /RET fail 2`] = `"select * from Estaciones where idEstacion = 'XXX'"`;

exports[`restServers/estaciones get /RET success 1`] = `Object {}`;

exports[`restServers/estaciones get /RET success 2`] = `"select * from Estaciones where idEstacion = 'RET'"`;

exports[`restServers/estaciones get /existe/ id /RET success 1`] = `Object {}`;

exports[`restServers/estaciones get /existe/ id /RET success 2`] = `"select count(*) as hay from Estaciones where idEstacion = 'RET'"`;

exports[`restServers/estaciones get /existe/ id /YYY fail 1`] = `Object {}`;

exports[`restServers/estaciones get /existe/ id /YYY fail 2`] = `"select count(*) as hay from Estaciones where idEstacion = 'YYY'"`;

exports[`restServers/estaciones get /existe/nombre /Retiro success 1`] = `Object {}`;

exports[`restServers/estaciones get /existe/nombre /Retiro success 2`] = `"select count(*) as hay from Estaciones where nombre = 'Retiro'"`;

exports[`restServers/estaciones get /existe/nombre /YYYY fail 1`] = `Object {}`;

exports[`restServers/estaciones get /existe/nombre /YYYY fail 2`] = `"select count(*) as hay from Estaciones where nombre = 'YYYY'"`;

exports[`restServers/estaciones get /idEstacion fail 1`] = `Object {}`;

exports[`restServers/estaciones get /idEstacion fail 2`] = `"select * from Estaciones where idEstacion = 'XXX'"`;

exports[`restServers/estaciones get /idEstacion success 1`] = `Object {}`;

exports[`restServers/estaciones get /idEstacion success 2`] = `"select * from Estaciones where idEstacion = 'RET'"`;

exports[`restServers/estaciones get /nombre/nombre fail 1`] = `Object {}`;

exports[`restServers/estaciones get /nombre/nombre fail 2`] = `"select * from Estaciones where nombre = 'Retiro'"`;

exports[`restServers/estaciones get /nombre/nombre success 1`] = `Object {}`;

exports[`restServers/estaciones get /nombre/nombre success 2`] = `"select * from Estaciones where nombre = 'Retiro'"`;

exports[`restServers/estaciones post duplicate 1`] = `Object {}`;

exports[`restServers/estaciones post duplicate 2`] = `"insert into Estaciones set ?"`;

exports[`restServers/estaciones post success 1`] = `Object {}`;

exports[`restServers/estaciones post success 2`] = `"insert into Estaciones set ?"`;

exports[`restServers/estaciones put /RET success 1`] = `Object {}`;

exports[`restServers/estaciones put /RET success 2`] = `"update Estaciones set \`nombre\` = 'Retiro' where idEstacion = 'RET'"`;

exports[`restServers/estaciones put /XXX fail 1`] = `Object {}`;

exports[`restServers/estaciones put /XXX fail 2`] = `"update Estaciones set \`nombre\` = 'Retiro' where idEstacion = 'XXX'"`;
